stream/src/public_key/benches/connection.rs

```diff
@@ -0,0 +1,67 @@
+use chacha20poly1305::{ChaCha20Poly1305, KeyInit};
+use commonware_runtime::mocks;
+use commonware_stream::{public_key::Connection, Receiver, Sender};
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+use futures::executor::block_on;
+
+fn benchmark_sender(c: &mut Criterion) {
+    let msg = b"hello";
+    let max_message_size = 1024 * 1024; // 1MB
+
+    c.bench_function(
+        &format!(
+            "max_message_size_len={} msg_len={}",
+            max_message_size,
+            msg.len()
+        ),
+        |b| {
+            b.iter_batched(
+                || {
+                    let cipher = ChaCha20Poly1305::new(&[0u8; 32].into());
+                    let (sink, stream) = mocks::Channel::init();
```

Comment:
I wonder if we should implement a `dummy::Channel` here that just drops messages on the ground?

Otherwise, I worry that the majority of time will be spent managing `mocks::Channel`?

Reply:
Just for sending? Because we do need the actual messages in the stream for `receive`, right? As decoding is part of the process?

Reply:
I guess its worth aligning on exactly what we are testing here.

I suppose another way to look at benchmarking is to pave the way for future optimization and as long as the overhead is the same across runs, we are good.